packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx

```diff
@@ -2,11 +2,10 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import { TextFieldProps } from '@mui/material/TextField';
 import { unstable_useId as useId } from '@mui/utils';
+import { useTimeout } from '../../../hooks/utils/useTimeout';
 import { GridFilterInputValueProps } from './GridFilterInputValueProps';
 import { useGridRootProps } from '../../../hooks/utils/useGridRootProps';
 
-export const SUBMIT_FILTER_STROKE_TIME = 500;
```

Comment:
Interestingly, changing the debounce time exposes an issue with filtering similar to https://github.com/mui/mui-x/pull/9630

https://codesandbox.io/s/naughty-brahmagupta-4knjzf

https://github.com/mui/mui-x/assets/13808724/c008aa06-f28e-4989-a5c0-3768637cff2a

Reply:
I've solved it by removing the effect, which didn't seem to be actually useful. You can review again, lmk if you see a way that removing the effect could have negative consequences.